src/standalone/Soil/energy_hydrology.jl

```diff
@@ -492,17 +491,22 @@ function ClimaLand.surface_resistance(
     p,
     t,
 ) where {FT}
-    return ClimaLand.Domains.top_center_to_surface(
-        ClimaLand.Soil.soil_resistance.(
-            p.soil.θ_l,
-            Y.soil.ϑ_l,
-            Y.soil.θ_i,
-            model.parameters,
-        ),
+    cds = ClimaCore.Fields.coordinate_field(model.domain.space.subsurface)
+    (; ν, θ_r, hydrology_cm) = model.parameters
+    S_l = effective_saturation.(ν, p.soil.θ_l, θ_r)
+    S_l_sfc = p.soil.sfc_scratch
+    ClimaLand.Domains.linear_interpolation_to_surface!(S_l_sfc, S_l, cds.z)
+    θ_l_sfc = @. S_l_sfc * (ν - θ_r) + θ_r
+    θ_i_sfc = ClimaLand.Domains.top_center_to_surface(Y.soil.θ_i)
```

Comment:
Maybe we can change this to use `linear_interpolation_to_surface` now too, since we know that function isn't allocating a new field. Or we could have a separate PR to go through and replace `top_center_to_surface` with `linear_interpolation_to_surface!` throughout the code

Reply:
I will change it here, and then open an issue for the other ones

Reply:
oh wait. it will be a different issue. because we are in the same function, but we only have one allocated scratch spot to hold the values (instead of alllocating), we cant reuse it in the same function. I think we need multiple surface scratch fields...

Reply:
noted here: https://github.com/CliMA/ClimaLand.jl/issues/615